### DISCUSSION TITLE
Failed to fetch certificate

### DISCUSSION THREAD

**Discussion by ismaelct** [epoptes/epoptes#143] [Q&A]
I'm on Arch.

140514673530240:error:0200206F:system library:connect:Connection refused:crypto/bio/b_sock2.c:110:
140514673530240:error:2008A067:BIO routines:BIO_connect:connect error:crypto/bio/b_sock2.c:111:connect:errno=111
epoptes-client ERROR: Failed to fetch certificate from 172.26.110.110:789

![image](https://user-images.githubusercontent.com/16766082/138055240-9d0e60a8-ecd5-48f7-a951-7457e4a4a958.png)

**Comment by alkisg** [comment]
Go to the epoptes-server PC at 172.26.110.110, run the following commands, and upload the output here:

```shell
ls -l /etc/epoptes
ps faux|grep epoptes
```

**Comment by ismaelct** [comment]
![20211020_103948.jpg](https://user-images.githubusercontent.com/16766082/138059053-1c109575-b1de-4bdd-9b1b-f1474b6f03a7.jpg)

**Comment by alkisg** [comment]
Check if you have a firewall running. If so, try to temporarily disable it.

Other ways to test for the ability to access certain ports on certain servers are mentioned there:
https://stackoverflow.com/questions/27312068/openssl-socket-connection-refused-connecterrno-111

**Comment by ismaelct** [comment]
On the client or the server?, i mean i tried to connect in ubuntu with no problem, it's just in my arch computer

**Comment by alkisg** [comment]
A firewall can block the connection either on the client or on the server, so you'd need to check both of them.
But if it works with your Ubuntu client and it doesn't work with the Arch client, then I would guess there's a firewall in the Arch client.

**Comment by ismaelct** [comment]
i tried to allow it tro ufw but didnt work so..
![image](https://user-images.githubusercontent.com/16766082/138070718-ffed0093-8a2e-4a9e-a42c-ad255f9abc6c.png)

**Comment by alkisg** [comment]
Try `sudo ufw disable`, then try to fetch the certificate.

**Comment by alkisg** [comment]
@ismaelct, I converted this issue into a discussion as it was not a bug in the epoptes code.
